xwiki-pro-macros-test/xwiki-pro-macros-test-pageobjects/src/main/java/com/xwiki/pro/test/po/generic/StatusMacroPage.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.pro.test.po.generic;
+
+import java.util.List;
+
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.xwiki.test.ui.po.ViewPage;
+
+public class StatusMacroPage extends ViewPage
+{
+    @FindBy(css = "span.statusBox")
+    public List<WebElement> statusBox;
+
+    public String getStatusTitle(int i)
+    {
+        return statusBox.get(i).getAttribute("title");
+    }
+
+    public int getStatusCount()
```

Comment:
Checking the size is a recurrent thing you test. Since I see the majority of the pages you have created will become actual `BaseElement` page objects (haven't checked yet all ob them, maybe some will remain as pages depending on what is needed), you can still have a `MacroPage` / `GenericMacrosPage` which can be used for all tests, and have it hold generic methods, like this one